File: persephone/provision-service/test/jvm/com/vmware/blockchain/deployment/service/provision/TestOrchestratorFactory.java
/* **************************************************************************
 * Copyright (c) 2019 VMware, Inc.  All rights reserved. VMware Confidential
 * *************************************************************************/
package com.vmware.blockchain.deployment.service.provision;

import java.util.concurrent.CompletableFuture;

import com.vmware.blockchain.deployment.model.orchestration.OrchestrationSiteInfo;
import com.vmware.blockchain.deployment.orchestration.Orchestrator;

/**
 * Test implementation of {@link OrchestratorProvider}.
 */
class TestOrchestratorFactory implements OrchestratorProvider {

    @Override
    public CompletableFuture<Orchestrator> newOrchestrator(OrchestrationSiteInfo site) {
        return CompletableFuture.completedFuture(new BookKeepingStubOrchestrator(site));
    }
}

File: persephone/provisioning-service/src/test/java/com/vmware/blockchain/services/provision/TestOrchestratorFactory.kt
/* **************************************************************************
 * Copyright (c) 2019 VMware, Inc. All rights reserved. VMware Confidential
 * **************************************************************************/
package com.vmware.blockchain.services.provision

import com.vmware.blockchain.deployment.orchestration.Orchestrator
import com.vmware.blockchain.deployment.orchestration.OrchestratorProvider
import com.vmware.blockchain.deployment.v1.OrchestrationSiteInfo

/**
 * Test implementation of [OrchestratorProvider].
 */
internal class TestOrchestratorFactory(
    private val validator: com.vmware.blockchain.services.provision.OrchestrationSiteValidator
) : OrchestratorProvider {

    override fun newOrchestrator(site: OrchestrationSiteInfo): Orchestrator {
        return com.vmware.blockchain.services.provision.BookKeepingStubOrchestrator(site, validator)
    }
}
